frontend/src/components/pages/UserManagement/index.tsx

```diff
@@ -15,68 +19,203 @@ import {
+    },
+    {
+      value: "donors",
+      label: "Donors",
+    },
+  ];
+
+  // Merge donors and volunteers into one list of type UserMgmtTableRecord
+  const buildTableUsers = (): UserMgmtTableRecord[] => {
+    const mergedUsers: UserMgmtTableRecord[] = [];
+
+    volunteers.forEach(function (volunteer) {
+      mergedUsers.push({
+        userId: volunteer.userId,
+        id: volunteer.id,
+        pointOfContact: `${volunteer.firstName} ${volunteer.lastName}`,
+        company: "",
+        email: volunteer.email,
+        phoneNumber: volunteer.phoneNumber,
+        accountType: volunteer.role,
+        approvalStatus: volunteer.status,
+      });
+    });
+
+    donors.forEach(function (donor) {
+      mergedUsers.push({
+        userId: donor.userId,
+        id: donor.id,
+        pointOfContact: `${donor.firstName} ${donor.lastName}`,
+        company: donor.businessName,
+        email: donor.email,
+        phoneNumber: donor.phoneNumber,
+        accountType: donor.role,
+        approvalStatus: "",
+      });
+    });
+
+    return mergedUsers;
+  };
 
   React.useEffect(() => {
     const getDonors = async () => {
       const res = await DonorAPIClient.getAllDonors();
       setDonors(res);
     };
 
+    const getVolunteers = async () => {
+      const res = await VolunteerAPIClient.getAllVolunteers();
+      setVolunteers(res);
+    };
+
     getDonors();
-  }, []);
+    getVolunteers();
+  }, [dataChanged]);
+
+  // Update users whenever donors or volunteers have a change
+  React.useEffect(() => {
+    setUsers(buildTableUsers());
+  }, [donors, volunteers]);
+
+  const tableData = React.useMemo(() => {
+    let filteredUsers;
+    if (selectedFilter === "all") filteredUsers = users;
+    else if (selectedFilter === "volunteers")
+      filteredUsers = users.filter(
+        (user) => user.accountType === Role.VOLUNTEER,
+      );
+    else
+      filteredUsers = users.filter((user) => user.accountType === Role.DONOR);
+
+    if (!search) return filteredUsers;
+    return users.filter(
+      (user) =>
+        user.pointOfContact.toLowerCase().includes(search.toLowerCase()) ||
+        user.company.toLowerCase().includes(search.toLowerCase()) ||
+        user.email.toLowerCase().includes(search.toLowerCase()) ||
+        user.phoneNumber.toLowerCase().includes(search.toLowerCase()),
+    );
+  }, [search, selectedFilter, users]);
+
+  // Sets volunteer status to Approved
+  const handleApprove = (user: UserMgmtTableRecord) => {
+    const newVolunteerData = { status: Status.APPROVED };
+    const updatedVolunteer = VolunteerAPIClient.updateVolunteerById(
+      user.id,
+      newVolunteerData,
+    );
+    setDataChanged(dataChanged === 0 ? 1 : 0);
+    console.log(dataChanged);
+  };
+
+  // Deletes selected user
+  const handleDeleteUser = (user: UserMgmtTableRecord) => {
+    const updatedUser = UserAPIClient.deleteUserById(user.userId);
+    setDataChanged(dataChanged === 0 ? 1 : 0);
+  };
```

Comment:
Calling `UserAPIClient.deleteUserById(user.userId)` for now, all of the delete stuff should be handled in this function @Shehryar21

Reply:
Yeah thats all that needs to be done here.